--- squashfs-tools/mksquashfs.c
+++ squashfs-tools/mksquashfs.c
@@ -109,6 +109,9 @@
 int use_regex = FALSE;
 int nopad = FALSE;
 int exit_on_error = FALSE;
+#if (TARGET_FORMAT == AVM_BE || TARGET_FORMAT == AVM_LE)
+int use_all_avm_changes_to_standard_format = FALSE;
+#endif
 
 long long global_uid = -1, global_gid = -1;
 
@@ -4461,6 +4464,10 @@
 			sBlk->lookup_table_start);
 
 	sBlk->bytes_used = bytes;
+#if (TARGET_FORMAT == AVM_BE || TARGET_FORMAT == AVM_LE)
+	if (use_all_avm_changes_to_standard_format)
+		sBlk->mkfs_time = (int) sBlk->bytes_used;
+#endif
 
 	sBlk->compression = comp->id;
 
@@ -4968,13 +4975,18 @@
 		} else if(strcmp(argv[i], "-no-duplicates") == 0)
 			duplicate_checking = FALSE;
 
+#if (TARGET_FORMAT == AVM_BE || TARGET_FORMAT == AVM_LE)
+		else if(strcmp(argv[i], "-avm-strict") == 0)
+			use_all_avm_changes_to_standard_format = TRUE;
+#endif
+
 		else if(strcmp(argv[i], "-no-fragments") == 0)
 			no_fragments = TRUE;
 
-		 else if(strcmp(argv[i], "-always-use-fragments") == 0)
+		else if(strcmp(argv[i], "-always-use-fragments") == 0)
 			always_use_fragments = TRUE;
 
-		 else if(strcmp(argv[i], "-sort") == 0) {
+		else if(strcmp(argv[i], "-sort") == 0) {
 			if(++i == argc) {
 				ERROR("%s: -sort missing filename\n", argv[0]);
 				exit(1);
@@ -5083,6 +5095,10 @@
 				"files larger than block size\n");
 			ERROR("-no-duplicates\t\tdo not perform duplicate "
 				"checking\n");
+#if (TARGET_FORMAT == AVM_BE || TARGET_FORMAT == AVM_LE)
+			ERROR("-avm-strict\t\tuse custom AVM format strictly"
+				"\n");
+#endif
 			ERROR("-all-root\t\tmake all files owned by root\n");
 			ERROR("-force-uid uid\t\tset all file uids to uid\n");
 			ERROR("-force-gid gid\t\tset all file gids to gid\n");

--- squashfs-tools/unsquashfs.c
+++ squashfs-tools/unsquashfs.c
@@ -1626,8 +1626,21 @@
 
 void squashfs_stat(char *source)
 {
-	time_t mkfs_time = (time_t) sBlk.s.mkfs_time;
-	char *mkfs_str = ctime(&mkfs_time);
+	char *mkfs_str = NULL;
+	int do_interpret_mkfs_time_as_time = 1;
+
+#if (TARGET_FORMAT == AVM_BE || TARGET_FORMAT == AVM_LE)
+	if (sBlk.s.s_major == 4 && sBlk.s.mkfs_time == (int)sBlk.s.bytes_used) {
+		do_interpret_mkfs_time_as_time = 0;
+		mkfs_str = "is not available because of modified AVM-format (mkfs_time == bytes_used)\n";
+	}
+#endif
+
+	if (do_interpret_mkfs_time_as_time) {
+		time_t mkfs_time = (time_t) sBlk.s.mkfs_time;
+		mkfs_str = ctime(&mkfs_time);
+	}
+
 
 	printf("Found a valid %sSQUASHFS %d:%d superblock on %s.\n",
 #if __BYTE_ORDER == __BIG_ENDIAN
