--- squashfs-tools/mksquashfs.c
+++ squashfs-tools/mksquashfs.c
@@ -5095,10 +5095,6 @@
 				"files larger than block size\n");
 			ERROR("-no-duplicates\t\tdo not perform duplicate "
 				"checking\n");
-#if (TARGET_FORMAT == AVM_BE || TARGET_FORMAT == AVM_LE)
-			ERROR("-avm-strict\t\tuse custom AVM format strictly"
-				"\n");
-#endif
 			ERROR("-all-root\t\tmake all files owned by root\n");
 			ERROR("-force-uid uid\t\tset all file uids to uid\n");
 			ERROR("-force-gid gid\t\tset all file gids to gid\n");
@@ -5144,6 +5140,10 @@
 			ERROR("\t\t\tOptionally a suffix of K, M or G can be"
 				" given to specify\n\t\t\tKbytes, Mbytes or"
 				" Gbytes respectively\n");
+#if (TARGET_FORMAT == AVM_BE || TARGET_FORMAT == AVM_LE)
+			ERROR("\nCustom options: (by Freetz project and other authors)\n");
+			ERROR("-avm-strict\t\tuse custom AVM format strictly\n");
+#endif
 			ERROR("\nMiscellaneous options:\n");
 			ERROR("-root-owned\t\talternative name for -all-root"
 				"\n");
--- squashfs-tools/unsquashfs.c
+++ squashfs-tools/unsquashfs.c
@@ -2917,6 +2917,7 @@
 				"regular expressions\n");
 			ERROR("\t\t\t\trather than use the default shell "
 				"wildcard\n\t\t\t\texpansion (globbing)\n");
+			ERROR("\nCustom options: (by Freetz project and other authors)\n");
 			ERROR("\t-exit-on-error\t\ttreat normally ignored errors as fatal\n");
 			ERROR("\t-scan or -k\t\ttreat filesystem as a combined image\n");
 			ERROR("\t\t\t\t(kernel+SquashFS) and scan it to locate the superblock\n");
